### PR TITLE
Mention contributors in pack readme

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37322

## Description
Add to pack file: "CONTRIBUTORS.md"
Pack README will be appended with this file's content, automatically, at the create artifacts step.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py [hunk 1]
```diff
@@ -7,8 +8,26 @@
 
 
 class Readme(TextObject):
-    def __init__(self, path: Union[Path, str]):
+    def __init__(self, path: Union[Path, str], pack_path: Union[Path, str] = ''):
+        self._path = path
+        self._pack_path = pack_path
```
Comment (dantavori) [inline demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py:13]: no need to pass `pack_path`:
```suggestion
        self._pack_path = os.path.dirname(path)
```

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/CONTRIBUTORS.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#### This pack was co-authored by:
+ - Contributor1
+ - Contributor2
+
+Thank you for contributing to Cortex XSOAR.
+
+If you are interested in contributing, visit our contribution process [here](https://xsoar.pan.dev/docs/contributing/contributing).
```
Comment (dantavori) [inline demisto_sdk/tests/test_files/content_slim/Packs/Sample01/CONTRIBUTORS.md:7]: do we want these lines and the title be modifiable by the contributor, or maybe we should concatenate them in `mention_contributors_in_readme()`?
@ShahafBenYakir wdyt?

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/readme/readme_test.py [hunk 3]
```diff
@@ -21,3 +21,18 @@ def test_objects_factory(file: Path):
 def test_prefix():
     obj = Readme(README)
     assert obj.normalize_file_name() == README.name
+
+
+def test_dump():
```
Comment (dantavori) [inline demisto_sdk/commands/common/content/tests/objects/pack_objects/readme/readme_test.py:26]:
```suggestion
def test_mention_contributors_in_readme():
```

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/readme/readme_test.py [hunk 4]
```diff
@@ -21,3 +21,18 @@ def test_objects_factory(file: Path):
...
+    """
+    Given: pack README
+
+    When: creating artifacts, if CONTRIBUTORS file exist in the pack
+
+    Then: Thanks to contributors
```
Comment (dantavori) [inline demisto_sdk/commands/common/content/tests/objects/pack_objects/readme/readme_test.py:32]:
```suggestion
    Then: Ensure credit to contributors is appended to the readme file
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 * Fixed an issue where the **Assume yes** flag did not work properly for some entities in the **format** command.
-* Inproved the error messages for separators in folder and file names in the **validate** command.
+* Improved the error messages for separators in folder and file names in the **validate** command.
+* Added the ability to append pack README with credit to contributors.
```
Comment (dantavori) [inline CHANGELOG.md:4]:
```suggestion
* Added the ability to append to the pack README credit to contributors.
```

**Review by dantavori — CHANGES_REQUESTED** [review]

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/CONTRIBUTORS.md (hunk 2 above)
Comment (daryakoval) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/CONTRIBUTORS.md:7]: We considered this option, but prefer doing this like it is now. 
We want to see this markdown file at Pack folder, with the messages.

**Comment by ShahafBenYakir** [comment]
Look ok to me

**Comment by daryakoval** [comment]
When there is no README: 
image

When readme exsits: 
image

**Review by dantavori — APPROVED** [review]
awesome 💪🏽

File: demisto_sdk/commands/create_artifacts/content_artifacts_creator.py [hunk 6]
```diff
@@ -615,7 +615,10 @@ def dump_pack(artifact_manager: ArtifactsManager, pack: Pack) -> ArtifactsReport
         if is_feed_pack and 'TIM' not in pack.metadata.tags:
             pack.metadata.tags.append('TIM')
         pack.metadata.dump_metadata_file(artifact_manager.content_packs_path / pack.id)
-    if pack.readme:
+    if pack.readme or pack.contributors:
+        if not pack.readme:
+            readme_file = os.path.join(pack.path, 'README.md')
+            open(readme_file, 'a+').close()
         pack_report += ObjectReport(pack.readme, content_packs=True)
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/create_artifacts/content_artifacts_creator.py:622]:
```suggestion
    if pack.readme or pack.contributors:
        if not pack.readme:
            readme_file = os.path.join(pack.path, 'README.md')
            open(readme_file, 'a+').close()
        readme_obj = pack.readme
        readme_obj.contributors = pack.contributors
        pack_report += ObjectReport(readme_obj, content_packs=True)
```

File: demisto_sdk/commands/common/content/objects/abstract_objects/text_object.py [hunk 7]
```diff
@@ -57,3 +57,6 @@ def to_str(self):
             self._unserialize()
 
         return self._text
+
+    def dump(self, dest_dir: Optional[Union[Path, str]] = None) -> List[Path]:
+        return super().dump(dest_dir)
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/content/objects/abstract_objects/text_object.py:62]: Remove

**Pull request merged**